src/core/ensemble_scheduler.cc

```diff
@@ -658,33 +660,25 @@ EnsembleContext::CheckAndSetEnsembleOutput()
     }
 
     // Use the memory type of the memory block as preferred memory type
-    TRTSERVER_Memory_Type dst_memory_type;
+    TRTSERVER_Memory_Type dst_memory_type, allocated_memory_type;
     int64_t memory_type_id;
     size_t content_size;
     memory_block->BufferAt(0, &content_size, &dst_memory_type, &memory_type_id);
 
     void* buffer;
+    int64_t device_id;
     RETURN_IF_ERROR(response_provider_->AllocateOutputBuffer(
         output_pair.first, &buffer, expected_byte_size, shape, dst_memory_type,
-        memory_type_id));
+        memory_type_id, &allocated_memory_type, &device_id));
 
     // Done with this output if 'expected_byte_size' is 0
     if (expected_byte_size == 0) {
       continue;
     } else if (buffer == nullptr) {
-      if (dst_memory_type != TRTSERVER_MEMORY_CPU) {
-        dst_memory_type = TRTSERVER_MEMORY_CPU;
-        memory_type_id = 0;
-        RETURN_IF_ERROR(response_provider_->AllocateOutputBuffer(
-            output_pair.first, &buffer, expected_byte_size, shape,
-            dst_memory_type, memory_type_id));
-      }
-      if (buffer == nullptr) {
-        return Status(
-            RequestStatusCode::INTERNAL,
-            "all attempts to allocate buffer for output '" + output_pair.first +
-                "' failed");
-      }
+      return Status(
```

Comment:
This isn't the right message anymore. There is only one allocation attempt. AllocateOutputBuffer and its callers need to be changed to this new way of working.

Reply:
Done. Will fix this message